examples/ensemble/plot_gradient_boosting_categorical.py

```diff
@@ -0,0 +1,212 @@
+# :class:`~preprocessing.OrdinalEncoder` to pre-process the data.
+#
+# The main difference between this pipeline and the previous one is that in
+# this one, we let the :class:`~ensemble.HistGradientBoostingRegressor` know
+# which features are categorical.
+
+# The ordinal encoder will first output the categorical features, and then the
+# continuous (passed-through) features
+categorical_mask = ([True] * n_categorical_features +
+                    [False] * n_numerical_features)
+hist_native = make_pipeline(
+    ordinal_encoder,
+    HistGradientBoostingRegressor(random_state=42,
+                                  categorical_features=categorical_mask)
+)
+
+
+# %%
+# Model comparison
+# ----------------
+# Finally, we evaluate the models using cross validation. Here we compare the
+# models performance in terms of
+# :func:`~metrics.mean_absolute_percentage_error` and fit times.
+
+from sklearn.model_selection import cross_validate
+import matplotlib.pyplot as plt
+
+scoring = "neg_mean_absolute_percentage_error"
+dropped_result = cross_validate(hist_dropped, X, y, cv=3, scoring=scoring)
+one_hot_result = cross_validate(hist_one_hot, X, y, cv=3, scoring=scoring)
+ordinal_result = cross_validate(hist_ordinal, X, y, cv=3, scoring=scoring)
+native_result = cross_validate(hist_native, X, y, cv=3, scoring=scoring)
+
+
+def plot_results(figure_title):
+    fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(12, 8))
+
+    plot_info = [('fit_time', 'Fit times (s)', ax1, None),
+                 ('test_score', 'Mean Absolute Percentage Error', ax2,
+                  (0, 0.20))]
+
+    x, width = np.arange(4), 0.9
+    for key, title, ax, y_limit in plot_info:
+        items = [dropped_result[key], one_hot_result[key], ordinal_result[key],
+                 native_result[key]]
+        ax.bar(x, [np.mean(np.abs(item)) for item in items],
+               width, yerr=[np.std(item) for item in items],
+               color=['C0', 'C1', 'C2', 'C3'])
+        ax.set(xlabel='Model', title=title, xticks=x,
+               xticklabels=["Dropped", "One Hot", "Ordinal", "Native"],
+               ylim=y_limit)
+    fig.suptitle(figure_title)
+
+
+plot_results("Gradient Boosting on Adult Census")
+
+# %%
+# We see that the model with one-hot-encoded data is by far the slowest. This
+# is to be expected, since one-hot-encoding creates one additional feature per
+# category value (for each categorical feature), and thus more split points
+# need to be considered during fitting. The native handling of categorical
+# features is slightly slower than treating categories as ordered quantities
+# ('Ordinal'), since native handling requires :ref:`sorting categories
+# <categorical_support_gbdt>`.
+#
+# In terms of prediction performance, dropping the categorical features leads
+# to poorer performance. The three models that use categorical features have
+# comparable error rates, with a slight edge for the native handling.
```

Comment:
I believe this is what this sentence is saying?

> The three models that use categorical features have
 comparable error rates, with a slight edge for the native handling

The time difference is also discussed right above

Reply:
I realize that it depends on the (CI) run. Now the native fits even faster than the ordinal one. In that case we should remove this statement *altogether*.

Reply:
Temporary proof: https://125629-843222-gh.circle-artifacts.com/0/doc/auto_examples/ensemble/plot_gradient_boosting_categorical.html

Reply:
I don't follow: this is about prediction performance, which is deterministic.

Reply:
nevermind I see it.

Reply:
Original point: We explicitly point out that fit time of ordinal is better, i.e.faster, than native: But we don't constrast this with the prediction performance, which is the reverse, i.e. native is better.

Now I realized that the statement about timing
> The native handling of categorical features is slightly slower than treating categories as ordered quantities ('Ordinal'), since ...

is dependent on the run. The current CI artifacts show that native fits faster than ordinal. This quoted statement should be removed.

Reply:
Sorry for the confusion.

Reply:
I expanded on this to note that this may not always be reflected in practice

Reply:
Also it looks like github has sent  "nevermind I see it" twice which makes me look like an ass but I only wrote this once lol (I sent it just as you sent your message)

Reply:
No offense taken 🤣  I guess the moment I approve and scroll down to the merge button, you will already have push it:smirk: